### ISSUE TITLE
How to assert against ApplicationDbContext

### ISSUE THREAD

**Issue by mattfrear** [aspnet/AspNetCore.Docs#8713]
Thank you for including the "Add a database context (ApplicationDbContext) using an in-memory database for testing." code block.

It would be useful to show how to access the system under test's ApplicationDbContext in order to assert say that a POST actually inserted a DB row with the correct data. The sample app contains no such test.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 77889d43-fedb-f8b4-8fea-379ec946555c
* Version Independent ID: f499b131-8b14-7e10-1728-543fdd5a5656
* Content: [Integration tests in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/test/integration-tests?view=aspnetcore-2.1#test-app-organization)
* Content Source: [aspnetcore/test/integration-tests.md](https://github.com/aspnet/Docs/blob/master/aspnetcore/test/integration-tests.md)
* Product: **aspnet-core**
* GitHub Login: @guardrex
* Microsoft Alias: **riande**

**Comment by Rick-Anderson** [comment]
@guardrex I think testing the DB is beyond the scope of this article.

**Comment by guardrex** [comment]
The work is already done ... waiting on approval from @javiercn. #9887 

I agree on the ask ... IMO, it would fall more directly to a unit test in most cases. However, AngleSharp makes checking a response so easy after a dB operation, that I thought it would be a nice add to the topic+sample to show how it would be set up in a simple case. Also, #9887 advances the sample to 2.2.

**Comment by guardrex** [comment]
Just for the record, this was at least temporarily dropped by engineering. I still have the bits sitting on the sidelines to resolve this issue. They'll triage this issue when they take a look at the Integration Testing project and decide what they'd like to do.

https://github.com/aspnet/Docs/projects/31

**Comment by Rick-Anderson** [comment]
@guardrex 	you can always blog the bits.

**Comment by bluebaroncanada** [comment]
I looked all over for a solution to this.  Figured I must've missed something important.  How can you even complete testing without this?

**Comment by mattfrear** [comment]
@bluebaroncanada I blogged how I do it here https://mattfrear.com/2018/10/10/verify-data-in-tests-with-asp-net-core-and-ef-core-in-memory/

**Comment by jepperaskdk** [comment]
> @bluebaroncanada I blogged how I do it here https://mattfrear.com/2018/10/10/verify-data-in-tests-with-asp-net-core-and-ef-core-in-memory/
> […](#)
> Sent from my phone
> On 20/04/2019, at 9:55 AM, bluebaroncanada ***@***.***> wrote: I looked all over for a solution to this. Figured I must've missed something important. How can you even complete testing without this? — You are receiving this because you authored the thread. Reply to this email directly, view it on GitHub, or mute the thread.

I can seed my database during `ConfigureServices` in my `CustomWebApplicationFactory` - however, the context is null using your solution: `Factory.Server.Host.Services.GetService(typeof(ApplicationDbContext)) as ApplicationDbContext`
It is throwing that is it not registered if using `GetRequiredService`. Any ideas?

**Comment by mattfrear** [comment]
> I can seed my database during `ConfigureServices` in my `CustomWebApplicationFactory` - however, the context is null using your solution: `Factory.Server.Host.Services.GetService(typeof(ApplicationDbContext)) as ApplicationDbContext`
> It is throwing that is it not registered if using `GetRequiredService`. Any ideas?

Dunno, something sounds weird with your DI setup.

Can you try `Factory.Server.Host.Services.GetService<ApplicationDbContext>()`? You might need the `Microsoft.Extensions.DependencyInjection.Abstractions` NuGet package to use that syntax.

Can you Get something else that isn't your DbContext? i.e. `Factory.Server.Host.Services.GetService<YourController>()`?

**Comment by Rick-Anderson** [comment]
Thanks for contacting us.
We don’t have the resources to invest in this area, so we are closing the issue. Should your request generate enough  👍 responses, we’ll reconsider.
